### PR TITLE
Feed a copy of model config in `RHO-LOSS` Twin Model

### PR DESCRIPTION
**Pull request by XianzheMa**

### CODE REVIEW

File: modyn/models/rho_loss_twin_model/rho_loss_twin_model.py [hunk 1]
```diff
@@ -26,8 +27,9 @@ def __init__(self, model_configuration: dict[str, Any], device: str, amp: bool)
         # we only need the inner model, not the wrapper
         self._models = nn.ModuleList(
             [
-                model_handler(rho_model_config, device, amp).model,
-                model_handler(rho_model_config, device, amp).model,
+                # some models change the model config dict during initialization
+                model_handler(copy.deepcopy(rho_model_config), device, amp).model,
+                model_handler(copy.deepcopy(rho_model_config), device, amp).model,
             ]
```
Comment (XianzheMa) [inline modyn/models/rho_loss_twin_model/rho_loss_twin_model.py:33]: for example, the ResNet models with setting `use_pretrained` to `True` https://github.com/eth-easl/modyn/blob/55dae262d066d88c64f303638ae96dbd02c0b5e7/modyn/models/resnet18/resnet18.py#L28

**Review by MaxiBoether — APPROVED** [review]
n🍨
